### ISSUE TITLE
S3 Gateway Does Not Cache

### ISSUE THREAD

**Issue by calclavia** [minio/minio#7869]
I'm running into a similar issue to https://github.com/minio/minio/issues/7325 when using Minio as a gateway to Amazon S3. I want to enable edge caching to avoid redownloading commonly used files by setting cache dir to `/mnt/cache`.

## Expected Behavior

I expect `/mnt/cache` to be not empty after downloading files from my S3 bucket through Minio server.

## Current Behavior

After downloading and uploading files, I notice the `/mnt/cache` folder is empty. I tested it with `du -sh /mnt/cache` when executing a shell within the Minio pod.

## Steps to Reproduce (for bugs)

1. Create an AWS bucket. Setup the AWS keys and permissions.
2. Helm install the official stable/minio chart with the following config. The access keys are redacted.
```
# Custom Minio config on top of the Helm charts
image:
  tag: RELEASE.2019-06-19T18-24-42Z

s3gateway:
  enabled: true
  replicas: 1
  serviceEndpoint: "https://s3.amazonaws.com"

persistence:
  enabled: false

accessKey: ...
secretKey: ...

environment:
  MINIO_ACCESS_KEY: ...
  MINIO_SECRET_KEY: ...
  
  MINIO_CACHE_DRIVES: "/mnt/cache"
  MINIO_CACHE_EXPIRY: "90"
  MINIO_CACHE_MAXUSE: "70"
```
3. Use `kubectl port-forward` to access Minio browser. Using the browser, download and upload some files.
4. Check `/mnt/cache` and notice it is empty.

## Context

See above.

## Regression

Not sure.

## Your Environment
Running Official Minio Helm charts on Kubernetes. See Helm chart config.

**Comment by poornas** [comment]
serviceEndpoint should point to the minio gateway here and not "https://s3.amazonaws.com"

ignore above, didnt realize this was helm chart.

**Comment by calclavia** [comment]
Any updates on this issue? Is this likely a bug or a misconfiguration on my end? Thanks!

**Comment by harshavardhana** [comment]
@calclavia did you not see @poornas update?  - also most probably to be fixed in #7694

**Comment by nitisht** [comment]
@calclavia just to keep you posted, https://github.com/helm/charts/pull/15749 will fix current issue you're seeing. It should be available soon

**Comment by calclavia** [comment]
@nitisht Thank you!

**Comment by harshavardhana** [comment]
Fixed by #7694
